save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/AgentsController.kt

```diff
@@ -57,9 +56,30 @@ class AgentsController(private val agentStatusRepository: AgentStatusRepository,
      */
     @GetMapping("/getAgentsStatusesForSameExecution")
     @Transactional
-    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> = agentRepository
-        .findByExecutionIdOfContainerId(agentId)
-        .map {
-            agentStatusRepository.findTopByAgentContainerIdOrderByTimeDesc(it.containerId)?.toDto()
+    fun findAllAgentStatusesForSameExecution(@RequestBody agentId: String): List<AgentStatusDto?> =
+            agentRepository.findAll { root, cq, cb ->
+                cb.equal(root.get<Execution>("execution"), getAgentByContainerId(agentId).execution)
+            }.map {
+                agentStatusRepository.findAll { root, cq, cb ->
+                    cq.orderBy(cb.desc(root.get<LocalDateTime>("time")))
+                    cb.equal(root.get<Agent>("agent").get<String>("containerId"), it.containerId)
+                }.get(0).toDto()
```

Comment:
You should limit the number of query results inside the query, not using clinet-side `get(0)`